File: snippets/typescript/protocol-client-quickstart_main.ts
import { AgentProtocolClient } from '@microsoft/agents-protocol';

// Create a client
const client = new AgentProtocolClient({
  baseUrl: 'https://agents.example.com/v1'
});

// Create and execute a run
const run = {
  agentId: 'agent_001',
  input: [
    {
      role: 'user',
      contents: [
        { type: 'text', text: 'Hello! Can you help me?' }
      ]
    }
  ]
};

// Wait for completion
const result = await client.runs.createAndWait(run);

console.log(`Status: ${result.status}`);
console.log(`Messages: ${result.messages.length}`);

File: snippets/python/protocol-client-quickstart_main.py
from microsoft.agents.protocol import AgentProtocolClient
import asyncio

async def main():
    # Create a client
    async with AgentProtocolClient.from_url("https://agents.example.com/v1") as client:
        # Create and execute a run
        run = {
            "agentId": "agent_001",
            "input": [
                {
                    "role": "user",
                    "contents": [
                        {"type": "text", "text": "Hello! Can you help me?"}
                    ]
                }
            ]
        }

        # Wait for completion
        result = await client.runs.create_and_wait(run)

        print(f"Status: {result['status']}")
        print(f"Messages: {len(result['messages'])}")

asyncio.run(main())
